bastionzero/service/agents/agents.go

```diff
@@ -0,0 +1,82 @@
+package agents
+
+import (
+	"context"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/service/targets"
+	"github.com/bastionzero/bastionzero-sdk-go/bastionzero/types"
+	"github.com/bastionzero/bastionzero-sdk-go/internal/client"
+	"net/http"
+)
+
+const (
+	agentsBasePath = "api/v2/agents"
+)
+
+// TargetsService handles communication with the agents endpoints of the
+// BastionZero API.
+//
+// BastionZero API docs:
+//
+// https://cloud.bastionzero.com/api/#tag--Agents
+
+type AgentsService client.Service
+
+type AgentDetails struct {
+	// Agent type.
+	AgentType AgentType `json:"type"`
+	// Last reported status of the agent.
+	AgentStatus AgentStatus `json:"status"`
```

Comment:
This will change to `agentstatus.AgentStatus` after making the package change mentioned in another comment